test/specs/babel-eslint.js

```diff
@@ -13,12 +13,15 @@ function lookup(obj, keypath, backwardsDepth) {
     return obj;
   }
 
-  return keypath
-    .split(".")
-    .slice(0, -1 * backwardsDepth)
-    .reduce((base, segment) => {
-      return base && base[segment], obj;
-    });
+  let pathSegments = keypath.split(".");
+
+  if (pathSegments.length > 1) {
+    pathSegments = pathSegments.slice(0, -1 * backwardsDepth);
+  }
+
+  return pathSegments.reduce((base, segment) => {
+    return base && base[segment], obj;
```

Comment:
Doesn't this end up being `return obj;`, meaning this whole statement is basically

```
return pathSegments.length === 1 ? pathSegments[0] : obj;
```

Reply:
Yeah, I think so. To be honest, I don't really feel like this `lookup` function provides a whole lot. The error message without modification already reads `Error: At loc.start.line: are different (1 !== 2)`. The extra logic just appends 
```
:
espree:
  'loc'
babel-eslint:
  'loc'
```
to it. Maybe we can just remove that code entirely?

Reply:
Yeah, could be great to switch to Jest! I'll make a note of that [here](https://github.com/babel/babel-eslint/issues/739). For now, would you be opposed to just removing this logic from the error message? If so, I can refactor it to the code snippet @loganfsmyth suggested above so we can land it!

Reply:
I'm not opposed to remove!

Reply:
Updated